File: C++/171.excel-sheet-column-number.cpp
/*
 * @lc app=leetcode id=171 lang=cpp
 *
 * [171] Excel Sheet Column Number
 */

// @lc code=start
#include <string>

using std::string;

class Solution
{
public:
    int titleToNumber(string columnTitle)
    {
        int total = 0;
        for (int i = 0; i < columnTitle.size(); i++)
        {
            total = 26 * total + (columnTitle[i] - 'A' + 1);
        }

        return total;
    }
};
// @lc code=end

File: C#/171.excel-sheet-column-number.cs
/*
 * @lc app=leetcode id=171 lang=csharp
 *
 * [171] Excel Sheet Column Number
 */

// @lc code=start
public class Solution {
    public int TitleToNumber(string columnTitle) {
        int result = 0;
        for (int i = 0; i < columnTitle.Length; i++) {
            result = result * 26 + (columnTitle[i] - 'A' + 1);
        }
        return result;
    }
}
// @lc code=end

